### DISCUSSION TITLE
Get-DBaComputerSystem Pipeline sql result

### DISCUSSION THREAD

**Discussion by JP-MB** [dataplat/dbatools#7441] [Q&A] [answered]
the following command doesn't success
 Invoke-DbaQuery -SqlInstance 'xxxxx\SQLS02'  -Query 'SELECT distinct top 1 [ComputerName] FROM [DBA_Maintenance].[dbo].[Repos_instances-prod-NLVCP]'|Get-DbaComputerSystem
but the following  yes: 
echo WWIN0281P | Get-DbaComputerSystem

What is the issue with data produced by sql request . Which step to introduce between the sql result and pipeline ?
rgds 
JP

**Comment by JP-MB** [comment]
PS C:\Users\a-xxxxxx> Invoke-DbaQuery -SqlInstance 'xxxxx\SQLS02'  -Query 'SELECT distinct top 1 [ComputerName] FROM [DBA_Maintenance].[dbo].[Repos_instances-prod]'|echo |Get-DbaComputerSystem
Get-DbaComputerSystem : The input object cannot be bound to any parameters for the command either because the command does not take pipeline input or the input and its properties do not match any of the parameters that take pipeline input.
At line:1 char:161
+ ... ance].[dbo].[Repos_instances]'|echo |Get-DbaComputerSystem
+                                                     ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (System.Data.DataRow:PSObject) [Get-DbaComputerSystem], ParameterBindingException
    + FullyQualifiedErrorId : InputObjectNotBound,Get-DbaComputerSystem


PS C:\Users\a-xxxxxx> echo WWWIN028 |Get-DbaComputerSystem


ComputerName            : WWWINxxx
Domain                  : xxxxxxxxxxxxxx
DomainRole              : Member Server
IsHyperThreading        : True
IsSystemManagedPageFile : False
Manufacturer            : Microsoft Corporation
Model                   : Virtual Machine
NumberLogicalProcessors : 16
NumberProcessors        : 1
ProcessorCaption        : Intel64 Family 6 Model 79 Stepping 1
ProcessorMaxClockSpeed  : 2594
ProcessorName           : Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz
SystemFamily            :
SystemType              : x64-based PC
TotalPhysicalMemory     : 128,00 GB

**Comment by niphlod** [comment]
Get-DbaComputerSystem accepts a "rich" instance object or a plain string that can pipelined to it.
Alternatively, you can use the -ComputerName parameter.

Now, the output from the query is an **object** whose property ComputerName is the string you may pass to Get-DbaComputerSystem

I think you may fix it with
```
$result = invoke-dbaquery ..... -Query "select top 1 ComputerName..."
$result.ComputerName | Get-DbaComputerSystem
```

**Reply by JP-MB** [reply]
with -As SingleValue suggested by andreas it works fine in stream command. thank you for your suggestion . JP

**Comment by andreasjordan** [comment, marked as answer]
If your query returns just one column, use `-As SingleValue`: 
```
Invoke-DbaQuery ..... -Query "select top 1 ComputerName..." -As SingleValue | Get-DbaComputerSystem
```

**Comment by JP-MB** [comment]
Super Andreas , it works fine . thank you so much . JP
